client-mvp-04/src/components/login/loginView.test.js

```diff
@@ -25,10 +25,14 @@ describe('Login View', () => {
       />,
       { wrapper: BrowserRouter }
     );
+  });
 
-    expect(screen.getByTestId('login')).toBeInTheDocument();
+  test('Should display form with input and `Sign In` button', () => {
+    render(<LoginView />, { wrapper: BrowserRouter });
+    expect(screen.getByTestId('login-container')).toBeInTheDocument();
     expect(screen.getByTestId('login-form')).toBeInTheDocument();
     expect(screen.getByTestId('login-input')).toBeInTheDocument();
+    expect(screen.getByPlaceholderText('Enter your email')).toBeInTheDocument();
```

Comment:
Here is an example of places where you can use data-testid's, and get exactly what you want. We really care about the login form input tells the user to "enter your email". What happens when the site has two two places with this same input? These types of tests looking for a string can break down the line. You can use both tests in one to accomplish your goals. 

You can do something like data-testid="email-input", then your test can be something like expect(screen.getByTestId="email-input".toContain("Enter your email"). Then your tests are really specific to the testid and to the value that you want the page to contain. The input's CSS can change, it can move anywhere, but as long as it has the test anchor and text, then it's good to go.

Reply:
1. I have data-testid for this input and I check if it is on the page
https://github.com/OlgaBilogurova/VRMS/blob/802555ea92f8d394a072e2bdea9846560248b030/client-mvp-04/src/components/login/loginView.test.js#L34

2. Site could have two places with this same input because loginView.test.js tests just code inside loginView.js.

2.  This line of code wouldn't work because placeholder it's not a value it's an attribute of the input.
screen.getByTestId="email-input".toContain("Enter your email")